src/features/Sync/SyncOutManager.ts

```diff
@@ -4,147 +4,87 @@ import { DatabaseService } from '../../common/services/Database';
-        await this.databaseService.query(SET_SYNC_PASSWORD, [password]);
-        this.config.password = password; 
-    }
-
-    get host(): string { return this.config.host }
-
-    get port(): string { return this.config.port }
-
-    get loginPath(): string { return this.config.paths.login }
-    
-    get sensorPath(): string { return this.config.paths.sensor }
-
-    get temperatureLogPath(): string { return this.config.paths.temperatureLog }
-
-    get temperatureBreachPath(): string { return this.config.paths.temperatureBreach }
-
-    get loginUrl(): string { return `${this.host}:${this.port}/${this.loginPath}` }
-
-    get sensorUrl(): string { return `${this.host}:${this.port}/${this.sensorPath}` }
-
-    get temperatureLogUrl(): string { return `${this.host}:${this.port}/${this.temperatureLogPath}` }
+    public setLoginUrl = async (url: string): Promise<void> => this.databaseService.query(SET_LOGIN_URL, [url]);
 
-    get temperatureBreachUrl(): string { return `${this.host}:${this.port}/${this.temperatureBreachPath}` }
+    public setSensorUrl = async (url: string): Promise<void> => this.databaseService.query(SET_SENSOR_URL, [url]);
 
-    get username(): string { return this.config.username }
+    public setTemperatureLogUrl = async (url: string): Promise<void> => this.databaseService.query(SET_TEMPERATURE_LOG_URL, [url]);
 
-    get password(): string { return this.config.password }
+    public setTemperatureBreachUrl = async (url: string): Promise<void> => this.databaseService.query(SET_TEMPERATURE_BREACH_URL, [url]);
 
-    getAuthenticationBody = (): string => JSON.stringify({ username: this.config.username, password: this.config.password });
+    public setUsername = async (username: string): Promise<void> => this.databaseService.query(SET_USERNAME, [username]); 
 
-    getSyncBody = (logs: SyncLog[]): string => JSON.stringify(logs.map(log => JSON.parse(log.payload))); 
+    public setPassword = async (password: string): Promise<void> => this.databaseService.query(SET_PASSWORD, [password]);
 
-    login = async (): Promise<void> => {
-        axios.post(this.loginUrl, this.getAuthenticationBody(), { withCredentials: true });
+    public login = async (loginUrl: string, username: string, password: string): Promise<void> => {
+        axios.post(loginUrl, this.getAuthenticationBody(username, password), { withCredentials: true });
     }
 
-    syncSensors = async (logs: SyncLog[]): Promise<void> => {
-        axios.put(this.sensorUrl, this.getSyncBody(logs), { withCredentials: true });
+    public syncSensors = async (sensorUrl: string, logs: SyncLog[]): Promise<void> => {
+        axios.put(sensorUrl, this.getSyncBody(logs), { withCredentials: true });
     }
 
-    syncTemperatureLogs = async (logs: SyncLog[]): Promise<void> => {
-        axios.put(this.temperatureLogUrl, this.getSyncBody(logs), { withCredentials: true });
+    public syncTemperatureLogs = async (temperatureLogUrl: string, logs: SyncLog[]): Promise<void> => {
```

Comment:
I believe https://app.swaggerhub.com/apis-docs/msupply-foundation/ColdChain/1.0.1#/default/coldchain_temperatureLogV1 the endpoint is returned in the `valid` field, all the temperature logs which were saved - I think we should be returning the response and iterating over the valid logs and removing the sync out's according to those ids?

Reply:
Issue created (#93).